### ISSUE TITLE
Automatic new type of layer creation mechanism for TensorFlow

### ISSUE THREAD

**Issue by andreiliphd** [tensorflow/tensorflow#23945]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.12
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

IDEA: I want to offer you a method that put production of a new type of layers on an industrial scale.
STATUS: Idea came to my mind I am thinking about implementation and I want to know your opinion.

**Will this change the current api? How?**

API may change as to provide compatibility to a new feature.

**Who will benefit with this feature?**

APPLICATION:
1. Creation of new type of layers suited for particular industries and application.
2. Research work can be enhanced. We are turning a neural network into the classic mathematical language.

**Any Other info.**

PROBLEM: How to turn a neural network in mathematical formula suited for a creation of an absolutely new type of layer for TensorFlow. It might be used in TensorFlow, Keras,  PyTorch or anything. 

HOW: 
1, You trained your network for particular industry application and invested a lot of computational power to make it work. And this neural network work for solving a particular problem in a particular industry very well.
2. You freeze all layers in your neural network.
3. Let's suppose we have an MSE loss for a simplicity.
4. Then you start to training target(label) looking for INTEGRAL of a function of your neural network. It will produce a mathematical formula describing a neural network. Let's call ANTI GRAD algorithm.
5. After it's done you have a mathematical formula that can easily be turned into a new type of layer for particular industry application with a simple(you can tune simplicity) formula for Chemistry, Economics, Medicine.

COMMENTS:
It would require a team of programmers to make it work. So, I am looking for new opportunities and the team to make it happen.
